presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduSplitManager.java

```diff
@@ -59,4 +79,56 @@ public ConnectorSplitSource getSplits(
                 throw new IllegalArgumentException("Unknown splitSchedulingStrategy: " + splitSchedulingStrategy);
         }
     }
+
+    private static class AsyncDelegate
```

Comment:
We could rename this to KuduDynamicFilteringSplitSource.
Instead of the future, we could take the delegate split source and dynamic filter as parameters.
You could then implement the waiting logic in the methods here using the dynamic filter and delegate split source.

Reply:
Do you mean generate the delegate split source without waiting for dynamic filter? But it seems currently we do not have a way to add the dynamic filter to the delegate split source later.

Reply:
I was thinking of something along the lines of `DynamicFilteringPageSource` in MemoryPageSourceProvider but adapted to `ConnectorSplitSource` and implementing a timeout instead of blocking until completion.
I think your current way works too, but doing it that way could be cleaner.

Reply:
Modified